src/coord/rmr/rq.c

```diff
@@ -184,10 +33,9 @@ void RQ_Push(MRWorkQueue *q, MRQueueCallback cb, void *privdata) {
   q->sz++;
 
   uv_mutex_unlock(&q->lock);
-  uv_async_send(&q->async);
 }
 
-static struct queueItem *rqPop(MRWorkQueue *q) {
+queueItem *RQ_Pop(MRWorkQueue *q, uv_async_t* async) {
```

Comment:
I don't see why we need the API change here. Each queue still has a single handle and is owned by its io runtime context

Reply:
But the only async handle that should be passed is the one owned by the `MRWorkQueue`. Why is it considered "mixing the UV loop with the RQ"

Reply:
Well because now MRWorkQueue does not own one.